server.js

```diff
@@ -209,14 +212,77 @@ app.get('/leaderboard-table', function(req, res) {
 
     ft = true;
     shrt = false;
+    
+    if (req.query.fullTable == 'false')
+        ft = false;
+    if (req.query.longTable == 'false')
+        shrt = true;
+    
+    if(req.query.type === 'points'){
+        lb.leaderboard(req.session.user.id, shrt, req.query.type, function(leader) {
+            var html = pointsTable({
+                fullTable: ft,
+                shortTable: shrt,
+                leaderboard: leader,
+                userid: req.session.user.id
+            });
+    
+            return res.status(200).send(html);
+        });
+    } else if (req.query.type === 'overall') {
+        lb.leaderboard(req.session.user.id, shrt, req.query.type, function(leader) {
+            var html = leaderboardTable({
+                fullTable: ft,
+                shortTable: shrt,
+                leaderboard: leader,
+                userid: req.session.user.id
+            });
+    
+            return res.status(200).send(html);
+        });
+    } else if (req.query.type === 'accuracy') {
+        lb.leaderboard(req.session.user.id, shrt, req.query.type, function(leader) {
+            var html = accuracyTable({
+                fullTable: ft,
+                shortTable: shrt,
+                leaderboard: leader,
+                userid: req.session.user.id
+            });
+    
+            return res.status(200).send(html);
+        });
+    } else if (req.query.type === 'attempt') {
+        lb.leaderboard(req.session.user.id, shrt, req.query.type, function(leader) {
+            var html = attemptTable({
+                fullTable: ft,
+                shortTable: shrt,
+                leaderboard: leader,
+                userid: req.session.user.id
+            });
+    
+            return res.status(200).send(html);
+        });
+    }
+});
 
+/* Fetch and render the points table. Send HTML as response. */
+app.get('/points-leaderboard', function(req, res) {
+    if (!req.session.user) {
+        return res.redirect('/');
+    }
+
+    var ft, shrt;
+
+    ft = true;
+    shrt = false;
+    
     if (req.query.fullTable == 'false')
         ft = false;
     if (req.query.longTable == 'false')
         shrt = true;
 
     lb.leaderboard(req.session.user.id, shrt, function(leader) {
-        var html = leaderboardTable({
+        var html = pointsTable({
```

Comment:
misleading name. We dont have a points table we have a leaderboard table thats sorted by points, why do we even have a different table for this

Reply:
Good point I'll change the name (I'll also do the same for the others as they're also called table). As for why we have a diffrent table for poitns, Larry wanted a points table :sweat_smile: